mlir/lib/Analysis/Presburger/Matrix.cpp

```diff
@@ -432,4 +432,138 @@ MPInt IntMatrix::normalizeRow(unsigned row, unsigned cols) {
 
 MPInt IntMatrix::normalizeRow(unsigned row) {
   return normalizeRow(row, getNumColumns());
+}
+
+MPInt IntMatrix::determinant() {
+  unsigned r = getNumRows();
+  unsigned c = getNumColumns();
+  assert(r == c);
```

Comment:
add an assert message please

`assert(r == c && "determinant can only be calculated for square matrices!");`